src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,269 @@
+  import com.google.cloud.bigquery.Field;
+  import com.google.cloud.bigquery.FieldValue;
+  import com.google.cloud.bigquery.FieldValueList;
+  import com.google.cloud.bigquery.Schema;
+  import com.google.cloud.bigquery.TableId;
+  import com.google.cloud.bigquery.TableResult;
+  import com.google.gson.Gson;
+  import com.google.gson.JsonElement;
+  import com.google.gson.JsonObject;
+  import io.cdap.e2e.utils.BigQueryClient;
+  import io.cdap.e2e.utils.PluginPropertyUtils;
+  import io.cdap.plugin.utils.SalesforceClient;
+  import io.cdap.plugin.utils.enums.SObjects;
+  import org.junit.Assert;
+
+  import java.io.IOException;
+  import java.sql.Date;
+  import java.time.LocalTime;
+  import java.time.OffsetDateTime;
+  import java.time.format.DateTimeFormatter;
+  import java.util.ArrayList;
+  import java.util.Arrays;
+  import java.util.List;
+
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+  static Gson gson = new Gson();
+
+  public static boolean validateSalesforceToBQRecordValues(String objectName, String targetTable) throws
+    IOException, InterruptedException {
+    String uniqueRecordId = SalesforceClient.queryObjectId(objectName);
+
+    List<JsonObject> bigQueryResponse = new ArrayList<>();
+    List<Object> bigQueryRows = new ArrayList<>();
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    List<JsonObject> sObjectResponse;
+    sObjectResponse = SalesforceClient.queryObject(uniqueRecordId, objectName);
+    return compareSalesforceAndJsonData(sObjectResponse, bigQueryResponse, targetTable);
+  }
+
+  public static boolean validateBQToSalesforceRecordValues(String targetTable, String objectName) throws
+    IOException, InterruptedException {
+    String uniqueRecordId = SalesforceClient.queryObjectId(objectName);
+
+    List<JsonObject> bigQueryResponse = new ArrayList<>();
+    List<Object> bigQueryRows = new ArrayList<>();
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    List<JsonObject> sObjectResponse;
+    sObjectResponse = SalesforceClient.queryObject(uniqueRecordId, objectName);
+    return compareSalesforceAndJsonData(sObjectResponse, bigQueryResponse, targetTable);
+  }
+
+  public static boolean validateSalesforceMultiObjectToBQRecordValues() throws IOException, InterruptedException {
+    List<String> objectNames = Arrays.asList(SObjects.AUTOMATION_CUSTOM__C.value, SObjects.AUTOMATION_CUSTOM2__C.value);
+    List<String> targetTables = getTableByName();
+    if (objectNames.size() != targetTables.size()) {
+      throw new IllegalArgumentException("Number of object names and target tables must be the same.");
```

Comment:
I believe instead of throwing and exception here we can use an Assertion.

Reply:
done